src/oidcop/oidc/add_on/pkce.py

```diff
@@ -41,7 +39,12 @@ def post_authn_parse(request, client_id, endpoint_context, **kwargs):
     :param kwargs:
     :return:
     """
-    if endpoint_context.args["pkce"]["essential"] and "code_challenge" not in request:
+    client = endpoint_context.cdb[client_id]
+    if "pkce_essential" in client:
+        essential = client["pkce_essential"]
+    else:
+        essential = endpoint_context.args["pkce"]["essential"]
```

Comment:
`essential = endpoint_context.args["pkce"].get("essential", False)`

@nsklikas how does it sounds to you?

Reply:
It's set in https://github.com/IdentityPython/oidc-op/pull/115/files#diff-60c1d62c690192be5964946a0883424948797ac8682e8ef94d9a59a6c8414d82L135. So it will always be there.

Should I remove that line and add the `get`?

Reply:
Not so important neither urgent, only if it sounds good to you.
Yes explicit is better than implicit but an absent value (boolan) could be False by default.
That's my tastes, not mandatory for this project!